server/sonar-server/src/main/java/org/sonar/server/computation/scm/ScmInfo.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * SonarQube, open source software quality management tool.
+ * Copyright (C) 2008-2014 SonarSource
+ * mailto:contact AT sonarsource DOT com
+ *
+ * SonarQube is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * SonarQube is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.server.computation.scm;
+
+/**
+ * Represents the Scm information for a specific file.
+ */
+public interface ScmInfo {
+
+  /**
+   * Get younger ChangeSet of the file. Can never be null
+   */
+  Changeset getLatestChangeset();
+
+  /**
+   * Get ChangeSet of the file for given line
+   *
+   * @throws IllegalArgumentException if line is bigger than ChangeSet size
```

Comment:
I would write "if there is not Changeset for the specified line, ie. the line number in invalid (either less than 1 or > linecount of the file)" or something of the kind